### DISCUSSION TITLE
maxDuration is not work

### DISCUSSION THREAD

**Discussion by hyundooo** [vercel/community#6189] [Help] [answered]
### Summary

I want to change the maxDuration to 600s, but it does not work.

### Steps to Reproduce

this is my vercel.json file
![image](https://github.com/vercel/community/assets/32191943/32a7d577-0919-4222-bdfb-dcdebd72a614)

also, I wrote code in my api file.
![image](https://github.com/vercel/community/assets/32191943/bb548549-eac4-4a41-a0d3-057fa7c60d5c)

but when I check the function it shows 15s max duration.
![image](https://github.com/vercel/community/assets/32191943/ff387ac2-7021-4579-a182-d02a0a65806c)

how can I fix it? I'm using pro account. that has 15s default setting.

**Comment by amyegan** [comment, marked as answer]
Hi @hyundooo. It looks like you're using Next.js pages router rather than app router. 

In that case, you should replace `export const maxDuration=299;` with `export const config = { maxDuration: 299 };`. Please give that a try and let me know if it works for you 🙏

**Reply by hyundooo** [reply]
it works. Thank you.

**Comment by humblemodulo** [comment]
What's protocol when routing via TRPC endpoints?  I've tried `vercel.json` as well as inline `export`s in multiple locations and nothing seems to have any effect.  Also, silly question, but:  Is the `app` in the example `app/api/...` literal, or are we supposed to substitute some other value in there?
